### ISSUE TITLE
Allow Sonata Autocomplete to use advanced filters

### ISSUE THREAD

**Issue by fliespl** [sonata-project/SonataAdminBundle#6300]
It would be great to add option to pass select advanced filter option with autocomplete ajax requests (get-autocomplete-items).

Currently selected advanced filter is not passed to ajax controller causing it to not filter according to setup.

In this case I have selected (not equals), but it still works as "equals"
https://ss.codeone.pl/ss-2020-08-20_14-09-55-1597925395.png

**Comment by kirya-dev** [comment]
Hmm, it looks like you don't know how it works. not equals applies to data table results, but not for autocomplete results.

But if you want configure this behavior you need determine callback
![image](https://user-images.githubusercontent.com/30481080/90985157-c0b7c680-e582-11ea-8b80-7f3a0cb050f2.png)
Also ajax request dont contain selected equal type. Im just use EqualOperatorType::TYPE_NOT_EQUAL in anyway for demonstrate callback usage.

From me: I will try to test this and get a bug: Filter will return all results when filter property is integer (eg. id) and non integer value in searchText (cast problem, maybe needs is_numeric check. if fail must no results).

**Comment by fliespl** [comment]
@kirya-dev you are totally right - I must have had brain damage that day and confused advanced filters with what I had in mind :)

Either way - let me rephrase, what I would like to achieve:
- have filter called "user"
- add autocomplete selection of users
- add additional subfilter to allow me select which property should be searched (username, email, id, hash)

Of course I can achieve the same by adding multiple filters:
- User by email
- User by username
- User by ID

but from usability - it would be better in my opinion to have one filter with ability to select property being searched.

Does this explaination clarifies it?

**Comment by kirya-dev** [comment]
So somplets solution is use some prefix
For ex: 
`email: user1@example.com`
`id: 123`
`username: kirya-dev`

Code:
```PHP
        $datagridMapper->add(
            'user',
            ModelAutocompleteFilter::class,
            ['show_filter' => true,],
            null,
            [
                'property' => 'id',
                'minimum_input_length' => 1,
                'callback' => static function (AdminInterface $admin, string $property, $searchText) {
                    $datagrid = $admin->getDatagrid();

                    $searchTextParts = explode(':', $searchText);
                    if (count($searchTextParts) === 2 && in_array($searchTextParts[0], ['id', 'email', 'username'])) {
                        [$property, $searchText] = $searchTextParts;
                    }

                    $datagrid->setValue($datagrid->getFilter($property)->getFormName(), null, $searchText);
                }
            ]
        );
```

**Comment by fliespl** [comment]
@kirya-dev great idea 🥇

**Comment by kirya-dev** [comment]
Close issue?

**Comment by jordisala1991** [comment]
The third example could be a great addition to our docs, wdyt @sonata-project/contributors ? IMO it could fit under the cookbook section. It can also be improved to accept multiple conditions (similar to what you can do with the github search box)
